src/api/control/member.rs

```diff
@@ -0,0 +1,102 @@
+//! Member definitions and implementations.
+
+use failure::Fail;
+use hashbrown::HashMap;
+
+use crate::log::prelude::*;
+
+/// Error that can be returned by Control API.
+#[derive(Fail, Debug, PartialEq)]
+pub enum ControlError {
+    /// [`Member`] is not found in repository.
+    #[fail(display = "Not found member")]
+    NotFound,
+}
+
+/// ID of [`Member`].
+pub type Id = u64;
+
+/// Media server user with its ID and credentials.
+#[derive(Clone, Debug)]
+pub struct Member {
+    /// ID of [`Member`].
+    pub id: Id,
+    /// Credentials to authorize [`Member`] with.
+    pub credentials: String,
+}
+
+/// Repository that stores store [`Member`]s.
+pub struct MemberRepository {
+    pub members: HashMap<Id, Member>,
+}
+
+impl MemberRepository {
+    /// Returns [`Member`] by its ID.
+    pub fn get_member(&self, id: Id) -> Result<Member, ControlError> {
+        debug!("retrieve member by id: {}", id);
+        self.members
+            .get(&id)
+            .map(|member| member.to_owned())
+            .ok_or(ControlError::NotFound)
+    }
+
+    /// Returns [`Member`] by its credentials.
+    pub fn get_member_by_credentials(
```

Comment:
@Kirguir we can reduce code noise here with renaming:  
`get_member()` -> `get()`  
`get_member_by_credentials()` -> `get_by_gredentials()`

No sense to specify `_member` here, as we are dealing with `Members` repository already.